src/sentinel.c

```diff
@@ -44,6 +44,11 @@
 
 extern char **environ;
 
+#if USE_OPENSSL == 1 /* BUILD_YES */
+extern SSL_CTX *redis_tls_ctx;
+extern SSL_CTX *redis_tls_client_ctx;
+#endif
```

Comment:
maybe we can make a hack that works in reverse and it'll still mean TLS can be a module, even for sentinel.
it looks odd to me that sentinel supports connection modules, but not TLS.

what if we declare these globals in redis, by default set to NULL, and have the TLS module write into them?

if it does work (and i think it should), many of the changes of this commit can be dropped (and the remaining ones squashed into the right place)

Reply:
> maybe we can make a hack that works in reverse and it'll still mean TLS can be a module, even for sentinel. it looks odd to me that sentinel supports connection modules, but not TLS.
> 
> what if we declare these globals in redis, by default set to NULL, and have the TLS module write into them?
> 
> if it does work (and i think it should), many of the changes of this commit can be dropped (and the remaining ones squashed into the right place)

Hi @yossigo 
Could you please give me any hint about this?

Reply:
@oranagra The point is that Sentinel uses hiredis, hiredis uses OpenSSL. Unless we completely change that so Sentinel will use the connection infrastructure, Sentinel+TLS will still need to be compiled and linked against OpenSSL.

Reply:
yes, so it must link against TLS anyway, but listening to TLS can come from a module.
it means it supports the module, but it doesn't remove the dependency of openssl.
i still think it's more streamline (we use the module approach everywhere).
you think it's misleading?

Reply:
i synced with Yossi about this, and here's the summary:
* Even if sentinel will take TLS (for inbound connections) from a module, and keep using hiredis for outbound connections, it'll mean that the TLS module has to be an OpenSSL one. i.e. it'll not be able to support an s2n TLS module since hiredis would be incompatible with it.
* For distros to compile and ship redis with TLS as a module (rather than static), won't be beneficial, as long as sentinel is still hard-linked with OpenSSL.
* We argue that Sentinel can't support connection modules at all (not just TLS), since (for now), these only control the inbound connections, but not the outbound (that's currently handled by hiredis).

This leaves with two options:
1. when compiling redis with TLS as a module, just avoid building sentinel at all (leave sentinel completely unsupported in that build).
2. when compiling redis with TLS as a module, build sentinel with only basic TCP support and no TLS capability.

we choose option [2], which i think is what currently implemented in this PR, with two possible required changes:
1. i see you removed the execution of the sentinel test suite in this mode, but i think we should still execute it, just do that without TLS.
2. maybe we should revert some of the changes we did in server.c with regards to allowing sentinel to load modules (that applies to any modules, not just connection modules).

Reply:
@pizhenwei i see you added a piece of code to prevent the TLS module to be loaded into sentinel, but i was under the impression we agreed that we won't support any module (connection or other modules) in sentinel mode.
i.e. revert REDISMODULE_CTX_FLAGS_SENTINEL and maybe some other changes we made in server.c initialization order.

the reason is that currently, any connection module loaded into sentinel will only affect the incoming connections into sentinel, and not the outgoing ones (unlike what we have for cluster and replication), so i think it makes little sense to allow sentinel to load connection modules.
do you disagree?

Reply:
i did say that, didn't think of sentinel at the time, in fact AFAIK it doesn't use rio (doesn't save RDB files)
maybe some day we'll want to allow sentinel to load modules (that can add commands etc), but for the moment, the main reason we did that was to allow adding connection modules, and we just realized that's not really gonna work (i.e. it only works on incoming connections, not on outgoing ones).
so i think we should revert that change.